core/src/main/java/io/substrait/plan/RelRoot.java

```diff
@@ -0,0 +1,53 @@
+package io.substrait.plan;
+
+import io.substrait.relation.Rel;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class RelRoot {
```

Comment:
The pojo `Rel` currently all use `Value.Immutable`, so that the builder will return an immutable instance.  Do we need do similar thing here for `RelRoot` ?